include/base/QnTools/StatBin.hpp

```diff
@@ -0,0 +1,220 @@
+// Contact: l.kreis@gsi.de; ilya.selyuzhenkov@gmail.com
+// For a full list of contributors please see docs/Credits
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef QNTOOLS_STATBIN_H_
+
+#include <cmath>
+#include <vector>
+
+namespace Qn {
+
+class StatBin {
+ public:
+  enum class Weights { REFERENCE, OBSERVABLE };
+  StatBin() = default;
+  explicit StatBin(Stats &stats) {
+    mean_     = stats.GetStatistic().Mean();
+    variance_ = stats.GetStatistic().Variance();
+    weight_   = stats.GetStatistic().SumWeights();
+    auto samples = stats.GetReSamples();
+    samples.CalculateMeans();
+    sample_means_ = samples.GetMeans();
+    sample_weights_ = samples.GetWeights();
+  }
+
+  [[nodiscard]] double ErrorFromPropagation() const { return std::sqrt(variance_); }
+  [[nodiscard]] double ErrorFromBootstrap() const {
+    Statistic stats;
+    for (const auto &mean : sample_means_) {
+      stats.Fill(mean, 1.0);
+    }
+    return stats.MeanError();
+  }
+
+  [[nodiscard]] double Mean() const { return mean_; }
+  [[nodiscard]] double Weight() const { return weight_; }
+  [[nodiscard]] double Error() const { return ErrorFromBootstrap(); }
+  [[nodiscard]] double Variance() const { return variance_; }
+
+  friend StatBin Merge(const StatBin &, const StatBin &);
+  friend StatBin MergeBins(const StatBin &, const StatBin &);
+  friend StatBin operator+(const StatBin &, const StatBin &);
+  friend StatBin operator-(const StatBin &, const StatBin &);
+  friend StatBin operator*(const StatBin &, const StatBin &);
+  friend StatBin operator/(const StatBin &, const StatBin &);
+
+  friend StatBin operator/(const StatBin &, double);
+  friend StatBin operator*(const StatBin &, double);
+  friend StatBin operator*(double, const StatBin &);
+
+  friend StatBin Pow(const StatBin &, double);
+  friend StatBin Sqrt(const StatBin &);
+
+
+ private:
+  std::vector<double> sample_means_;
+  std::vector<double> sample_weights_;
+  double mean_ = 0.;
+  double variance_ = 0.;
+  double weight_ = 0.;
+
+  /// \cond CLASSIMP
+ ClassDef(StatBin, 4);
+  /// \endcond
+};
+
+inline StatBin Merge(const StatBin &lhs, const StatBin &rhs) {
+  StatBin merged;
+  // mean and variance
+  merged.weight_ = lhs.Weight() + rhs.Weight();
+  merged.mean_ = (lhs.Mean() * lhs.Weight() + rhs.Mean() * rhs.Weight()) / merged.Weight();
```

Comment:
if `lhs.Mean()` or `rhs.Mean()` is _nan_, `merged.mean_` gets _nan_.

Test case:
* merge two empty StatBin-s
* merge result with filled StatBin

Reply:
implemented a check based on empty weights